nvflare/private/fed/server/server_engine.py

```diff
@@ -95,32 +97,38 @@ def __init__(self, server, args, client_manager: ClientManager, snapshot_persist
 
         # self.job_def_manager = job_def_manager  # todo: need to figure out how to initialize job manager with inputs
         self.job_def_manager = SimpleJobDefManager(
```

Comment:
All of these objects should be created during the config processing: fl_server.json in startup and/or in the "site_config" folder (TBD).

There will need to be a new event type such as "START_SYSTEM". These system-wise components (vs. RUN-wise) will handle this event type and get themselves fully initialized (e.g. one StudyManager needs to get its Storage object).